src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1756,14 +1772,18 @@ private ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedA
                         {
                             // bind the argument against each applicable parameter
                             var unboundArgument = (UnboundLambda)argument;
-                            foreach (var parameterList in parameterListList)
+                            if (NestedLambdaExpressionDepth(unboundArgument.Syntax) <= MaxLambdaExpressionDepthForErrorRecovery &&
```

Comment:
>NestedLambdaExpressionDepth(unboundArgument.Syntax) <= MaxLambdaExpressionDepthForErrorRecovery [](http://example.com/codeflow?start=32&length=95)

This condition looks questionable to me. Perhaps, the lambda is in the third level of nesting. Why is that important on its own? We might not be rebinding any of the enclosing lambdas for error recovery, doing simple plain body binding, and recovering in this case will not cause any noticeable overhead.

The issue says: "Reduce the overhead of binding lambdas when there are nested lambda expressions with multiple overloads. Specifically, consider avoiding binding nested lambdas unnecessarily for overloads that do not match." It doesn't look like any of the conditions, except the nesting, are checked.